service/history/task/transfer_active_task_executor.go

```diff
@@ -209,7 +210,12 @@ func (t *transferActiveTaskExecutor) processActivityTask(
 		return errWorkflowRateLimited
 	}
 
-	return t.pushActivity(ctx, task, timeout, mutableState.GetExecutionInfo().PartitionConfig)
+	err = t.pushActivity(ctx, task, timeout, mutableState.GetExecutionInfo().PartitionConfig)
+	if err == nil {
+		scope := common.NewPerTaskListScope(domainName, task.TaskList, types.TaskListKindNormal, t.metricsClient, metrics.TransferActiveTaskActivityScope)
```

Comment:
is it cheap to create these per tasklist scopes on the fly and discard? this will be done per shard so there will be 16k x num_tasklists of these across hosts